src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CommonField.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.ccd.domain.model.aggregated;
+
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.COMPLEX;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.PREDEFINED_COMPLEX_ADDRESS_GLOBAL;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.PREDEFINED_COMPLEX_ADDRESS_GLOBAL_UK;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.PREDEFINED_COMPLEX_ADDRESS_UK;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.PREDEFINED_COMPLEX_CASELINK;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.PREDEFINED_COMPLEX_ORDER_SUMMARY;
+
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldType;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.commons.lang3.StringUtils;
+
+public interface CommonField {
+
+    FieldType getFieldType();
+
+    String getId();
+
+    List<AccessControlList> getAccessControlLists();
+
+    void setDisplayContext(String displayContext);
```

Comment:
if you mean to implement `markDisplayContextToReadonly()` as a default method, it won't work since we won't have any handle to `setDisplayContext(...)` method. This is intended to provide a handle to that method in a common way...

Reply:
not as a default method. Just a method declaration here and concrete class assign read-only value to context variable. Don't worry if it doesn't fir nicely.